Bump minimum MC/Forge versions to match build.properties

uecasm opened the pull request:
# Changes proposed in this pull request:
- Bump minimum MC/Forge versions to match build.properties
- Make mods.toml update automatically when bumping versions in the future
- Add optional mod dependencies

Review please

Tested by trying to build mcol for 1.16.4, but that failed due to something related to the zombie villager feature.  So I believe this limit is accurate and the toml shouldn't be over-promising.  But I admit I didn't try taking a prebuilt jar and trying to run it against 1.16.4 outside a dev environment, so it might not be impossible that this was intended.  (Although it looks like the CF upload stopped claiming 1.16.4 compatibility at 0.14.162.)

uecasm committed:
Bump minimum MC/Forge versions to match build.properties

uecasm commented on src/main/resources/META-INF/mods.toml:
```diff
@@ -41,7 +41,7 @@ MineColonies is a colony simulator within Minecraft! There are numerous types of
     mandatory=true #mandatory
     # The version range of the dependency
-    versionRange="[34.1.0,)" #mandatory
+    versionRange="[36.1.0,)" #mandatory
```
The build.properties refers to 36.1.24; wasn't sure if I should use that exact version as a minimum or if it should be more lenient.  This seemed like an appropriate compromise.

uecasm commented:
While I'm here, should I add the optional minimum for Tinkers as well?  Or should that be a separate PR?

Raycoms commented:
> While I'm here, should I add the optional minimum for Tinkers as well? Or should that be a separate PR?

You can add this in here

uecasm committed:
DRY

uecasm committed:
Include optional dependencies

uecasm committed:
JEI is also an optional dependency.
And Dynamic Trees uses a more annoying version format than previously expected.
